src/components/Login/utils.ts

```diff
@@ -11,15 +11,45 @@ interface DecodedQrResponse {
   key: string;
 }
 
-export const decodeQr = (code: string): DecodedQrResponse => {
+const isJSON = (str: string): boolean => {
   try {
-    const parsedCode: QrCode = JSON.parse(code);
-    if (!parsedCode.endpoint) throw new AuthInvalidError("No endpoint");
-    if (!parsedCode.key) throw new AuthInvalidError("No key");
-    return { endpoint: parsedCode.endpoint, key: parsedCode.key };
+    JSON.parse(str);
+    return true;
   } catch (e) {
-    if (e.message.includes("token") || e instanceof SyntaxError)
```

Comment:
noticed that the `decodeQr` function on line 38 no longer has a try catch, will this `SyntaxError` go undetected, or will the `isJSON` function cover this scenario?

Reply:
Previously when it was only JSON, we could check for syntax error as we know JSON must strictly begin with curly open brace, but now with the addition of URL string type, any syntax error or parsing error will result on `isJSON` being interpreted as a URL.

Instead of using error as a way to tell if the string is a JSON or URL, we could perhaps scan the entire string for `{` and `}` character, if these are present, it means that we can try to parse it as a JSON and throw syntax error upon failure. However, if no `{` or `}` is detected, then we parse it as URL.

Reply:
okay thanks for explaining, while this works, it seems counter-intuitive (e.g. invalid json shouldn't be detected as a URL) and `if (!parsedCode.endpoint && !parsedCode.key)` covering for `SyntaxError` seems more like a side effect rather than a clear intended outcome.

Reply:
good to resolve this or at least have some comments explaining how it works

Reply:
oh i see, hmm how about making the parser detect for curly brace character `{` in the QR string so that we know it is a JSON first then we can check if it is a syntax error.

Reply:
hmm that might not be foolproof as well, instead we can parse the QR and check if it's either a 1. JSON or 2. URL, if it's neither we can treat that is an invalid QR

current behaviour:
1. Check if JSON 
2. If not JSON, parse as URL and find endpoint and key
3. If endpoint and key not found, throw error (JSON syntax errors falls under this category, which could go undetected, we might think its a wrong URL)

Improved behaviour 
1. Check if JSON
2. If not JSON, parse and check if URL
3. If not URL, throw error (not URL and not JSON, SyntaxError covered here)